debezium-core/src/main/java/io/debezium/pipeline/source/spi/ChangeEventSource.java

```diff
@@ -30,19 +30,6 @@ interface ChangeEventSourceContext {
          */
         void waitSnapshotCompletion() throws InterruptedException;
 
-        /**
-         * Wait for the resumeStreaming function to be called, which indicates that a snapshot is done
-         * and that streaming should resume.
-         *
-         * @param heartbeatCallback A callback function which will be periodically called while waiting for
-         *                          the snapshot to be completed.  Implementations should be kept simple and
-         *                          relatively fast: only do activities like generating activity on the
-         *                          streaming database connection to prevent idle timeouts.  Implementations
-         *                          should also check their own ElapsedTimeStrategy to control how often any
-         *                          heartbeat activities actually occur.
-         */
-        void waitSnapshotCompletion(Runnable heartbeatCallback) throws InterruptedException;
```

Comment:
This could still be retained, if you think there is future value in having a periodic callback function while waiting for snapshot completion.  However, at the moment, it would be dead code if we kept it.